### DISCUSSION TITLE
My server doesnt work

### DISCUSSION THREAD

**Discussion by Derxibitor** [Pryaxis/TShock#2471] [Q&A]
So, i have just started a server with plugin Crossplay to play with my friend. My friend doesnt have a computer only mobile phone. It seems that my server works, plugin also wokrs, but my friend can not join the server by typing my IP address.

yeah, i knoh that server has to run on 1.4.0.5. i started on this vesion
i also convienced that plugin for crossplay works, because Tshock whote "Crossplay plugin initiated"

can somebody help me please?

plugin i used - https://github.com/Moneylover3246/Crossplay/releases/tag/8bac2da

**Comment by Arthri** [comment]
> my friend can not join the server by typing my IP address.

what happens when they do that? Do they get disconnected? What message does it say/get stuck on

**Reply by Derxibitor** [reply]
He cant even connect

**Reply by Arthri** [reply]
So... he gets stuck on "Connecting to [your ip]..."?

**Reply by Derxibitor** [reply]
yes, you are right

**Reply by Derxibitor** [reply]
> So... he gets stuck on "Connecting to [your ip]..."?

so what i should do

**Comment by Arthri** [comment]
Is the IP you're giving them `192.168.1.x`, `127.0.0.1`, or `localhost`? If so then that's the wrong IP

If not then put your IP through https://www.yougetsignal.com/tools/open-ports/

If it says port closed it means you didn't port forward your server correctly. Your server must be port forwarded to allow others to join

## Port Forwarding
Before proceeding with port forwarding, be aware that this is not safe and do it at your own risk. There are many guides out there and each router has its own process, I recommend you google it along with your router information. There's too many out there to provide a comprehensive guide for all of them

## Tunneling
Alternatively, if you can't port forward or don't want to, you can use the much safer tunnels.

[ngrok](https://ngrok.io) is one of them, but there are also others like([localhost.run](http://localhost.run))

**Reply by Derxibitor** [reply]
Ok, i'll try it

**Reply by Derxibitor** [reply]
sadly, it still doesnt work, he was typing my IP address. So what can i do?

**Reply by Arthri** [reply]
You port forwarded right? If you did it probably means: A. you didn't port forward properly, B. you can't port forward

If you instead tunneled, please read the ngrok/localhost.run documentation

**Reply by Derxibitor** [reply]
> You port forwarded right? If you did it probably means: A. you didn't port forward properly, B. you can't port forward
> 
> If you instead tunneled, please read the ngrok/localhost.run documentation

this thing gets on my nerves, i invite my another friend that plays on PC and he connected, but friend that plays on mobile phone not. It seems that plugin doesnt work with phones, i have already written to MoneyLover who posted this plugin. I looking forward to his answer.
Yes, when i started a server i choose forward yes. Do you know how to create a server that can connect mobile phones. Can you please explain step by step? I also can give you my discord, so it is more comfortable and i can show you my screen in real time.

**Reply by Arthri** [reply]
imo it is unlikely to have something to do with the plugin. If the plugin malfunctioned it would display another message, not get stuck at "Connecting to ..."

> Yes, when i started a server i choose forward yes.

that doesn't work btw

---

> Do you know how to create a server that can connect mobile phones.

I do however I am unable to provide a step by step guide at the moment. The alternative method requires you to have knowledge of Node.JS and some internal Terraria stuff. However if you do insist

https://github.com/popstarfreas/Dimensions
https://github.com/popstarfreas/Dimensions-mcl/tree/feature/1421

**Reply by Derxibitor** [reply]
Thank you so much, i really appreciate your help and information you give me. This information just impossible to find in internet, you also give me a step by step guide. When i back home, i'll try it and write you how it go
